aoc2022/src/main/scala/utils.scala

```diff
@@ -79,6 +87,28 @@ object Grid2D:
 
       go(obstacles.sortBy(_._1.toTuple.swap).distinct)
 
+  
+  case class Line(delta: Int, b: Int):
+    def fx: Int => Int = (x: Int) => delta * x + b
+    def fy: Int => Int = (y: Int) => (y - b) / delta
+    def fyBounded(min: Int, max: Int): Int => Option[Int] = 
+      (y: Int) => 
+        val x: Int = fy(y)
+        if x < min || x > max then None else Some(x)
```

Comment:
Perhaps `Option.when` is better readable, but maybe it is a matter of taste.

Reply:
Cool the `Option.when`. Didn't know if its existence yet.